src/Native/LDANative/lda_engine.cpp

```diff
@@ -0,0 +1,1103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <unordered_map>
+#include <cstdint>
+#include <string>
+#include <iostream>
+#include <cstdlib>
+#include <time.h>
+#include <mutex>
+#include <set>
+#include <fstream>
+#include <thread>
+#include <algorithm>
+#include <stdlib.h>
+
+#include "timer.h"
+#include "rand_int_rng.h"
+#include "lda_document.h"
+#include "data_block.h"
+#include "model_block.h"
+#include "lda_engine.hpp"
+#include "utils.hpp"
+#include "simple_barrier.h"
+#include "light_doc_sampler.hpp"
+
+#ifdef _MSC_VER
+#include "windows.h"
+#else
+#include "sched.h"
+#endif
+
+namespace lda {
+    LdaEngine::LdaEngine(int numTopic,
+        int numVocab,
+        float alphaSum,
+        float beta,
+        int numIter,
+        int likelihoodInterval,
+        int numThread,
+        int mhstep,
+        int maxDocToken)
+        : K_(numTopic),
+        V_(numVocab),
+        compute_ll_interval_(likelihoodInterval),
+        beta_(beta),
+        num_iterations_(numIter),
+        mh_step_(mhstep),
+        alpha_sum_(alphaSum),
+        maxDocToken_(maxDocToken),
+        samplers_(nullptr),
+        document_buffer_(nullptr)
+    {
+        if (numThread > 0)
+        {
+            num_threads_ = numThread;
+        }
+        else
+        {
+            unsigned int uNumCPU = std::thread::hardware_concurrency();
+            num_threads_ = std::max(1, (int)(uNumCPU - 2));
+        }
+        printf("using %d thread(s) to do train/test\n", num_threads_);
```

Comment:
>printf [](start = 8, length = 6)

can you create issue to strip out all std cout and printf from this code and replace them with Message function call?